### PR TITLE
Add flutter update packages to some tests.

### PR DESCRIPTION
**Pull request by godofredoc**
Analyzer_benchmark and flutter_test_performance require to pub get multiple directories. Rather than delegating the recipes to run flutter update-packages we are explicitly running on tests that need it.

Bug: https://github.com/flutter/flutter/issues/75524

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/devicelab/bin/tasks/analyzer_benchmark.dart [hunk 1]
```diff
@@ -4,7 +4,9 @@
 
 import 'package:flutter_devicelab/tasks/analysis.dart';
 import 'package:flutter_devicelab/framework/framework.dart';
+import 'package:flutter_devicelab/framework/utils.dart';
 
 Future<void> main() async {
+  await flutter('update-packages');
```
Comment (christopherfujino) [inline dev/devicelab/bin/tasks/analyzer_benchmark.dart:10]: if we're adding this into the specific test, can we explicitly call pub get in the directory this test needs?

Comment (christopherfujino) [reply dev/devicelab/bin/tasks/analyzer_benchmark.dart]: nvm, this benchmark seems to depend on most packages.

**Comment by godofredoc** [comment]
@christopherfujino this PR is ready for review.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by justinmc** [comment]
@godofredoc The build is broken on this PR because of an analyzer error, but I'm not sure if it's related...  Do you have any idea? https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8855549842472159072/+/steps/run_test.dart_for_flutter_plugins_shard_and_subshard_analyze/0/stdout

**Comment by godofredoc** [comment]
> build is broken on this PR be

The change in this PR was to run flutter update-packages for the analyzer test. The one failing is flutter_plugins, let me retry it to see if it was a flake.

**Comment by justinmc** [comment]
Looks like it's still failing with the same error.

**Comment by christopherfujino** [comment]
The failure is unrelated to this PR.

**Pull request merged**